### PR TITLE
Update python Dataflow API, and add FlexRS pipeline option.

### PR DESCRIPTION
**Pull request by edre**
Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](../.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py [hunk 1]
```diff
@@ -99,6 +99,34 @@ def test_set_subnetwork(self):
...
+
+    env = apiclient.Environment([], #packages
+                                pipeline_options,
+                                '2.0.0', #any environment version
+                                FAKE_PIPELINE_URL)
+    self.assertEqual(
+        env.proto.flexResourceSchedulingGoal,
+        (dataflow.Environment.FlexResourceSchedulingGoalValueValuesEnum.
+         FLEXRS_SPEED_OPTIMIZED))
+
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py]: Could you add one more test with option not set.

File: sdks/python/apache_beam/runners/dataflow/internal/clients/dataflow/__init__.py [hunk 2]
```diff
@@ -1,35 +1,10 @@
...
-# Protect against environments where apitools library is not available.
-# pylint: disable=wrong-import-order, wrong-import-position
-try:
-  from apitools.base.py import *
-  from apache_beam.runners.dataflow.internal.clients.dataflow.dataflow_v1b3_messages import *
-  from apache_beam.runners.dataflow.internal.clients.dataflow.dataflow_v1b3_client import *
-except ImportError:
-  pass
-# pylint: enable=wrong-import-order, wrong-import-position
+from apitools.base.py import *
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/clients/dataflow/__init__.py]: Could you revert the changes to this file?

File: sdks/python/apache_beam/runners/dataflow/internal/clients/dataflow/dataflow_v1b3_client.py [hunk 3]
```diff
@@ -1,28 +1,7 @@
...
-
 """Generated client library for dataflow version v1b3."""
 # NOTE: This file is autogenerated and should not be edited by hand.
-
-from __future__ import absolute_import
-
 from apitools.base.py import base_api
-
-from apache_beam.runners.dataflow.internal.clients.dataflow import dataflow_v1b3_messages as messages
+import dataflow_v1b3_messages as messages
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/clients/dataflow/dataflow_v1b3_client.py]: Please revert the changes to the header and imports.

File: sdks/python/apache_beam/runners/dataflow/internal/clients/dataflow/dataflow_v1b3_messages.py [hunk 4]
```diff
@@ -95,11 +76,15 @@ class ApproximateSplitRequest(_messages.Message):
   Fields:
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/clients/dataflow/dataflow_v1b3_messages.py]: Please revert the changes to the header and imports.

**Comment by edre** [comment]
Resolved comments.
